client/src/components/sources/index.tsx

```diff
@@ -40,24 +27,37 @@ const SourcesView = (): React.ReactElement => {
   const toaster = useSelector((state: IReduxStore) => state.toaster);
 
   const [sourceToDelete, setSourceToDelete] = React.useState(
-    undefined as Source | undefined
+    undefined as ISelectedSource | undefined
   );
   const [isAlertOpen, setIsAlertOpen] = React.useState(false);
 
   const { data: dataSources, loading: loadingSources } = useQuery(qSources, {
-    fetchPolicy: 'network-only'
+    fetchPolicy: 'cache-and-network'
   });
 
   const removeSourceFromCache = (
     cache: any,
     { data: { deleteSource } }: any
+  ): void => updateCachedSource(cache, deleteSource, true);
+
+  const updateCachedSource = (
+    cache: any,
+    source: ISelectedSource,
+    toDelete?: boolean
```

Comment:
A bit tricky but I understand that it's more compact to have this here.
I'm ok with that if we don't find anything better

Reply:
I prefer doing it here rather than duplicating this code in multiple components. I find it annoying that the apollo cache does not update automatically at every mutation, I think that we could use some plugin that updates the cache automatically instead of having to do it manually. I'll look into it when i'll find the time.